### PR TITLE
Notebook that generates mock data and matches mentees/mentors.

### PR DESCRIPTION
**Pull request by dankositzke**
Notebook that generates mock data and matches mentees/mentors. Also including a csv file that contains the most used programming languages along with their popularity percentages.

(https://www.loom.com/share/d2009590be5a4d4e9065049b5f2aefb6)

### CODE REVIEW

**Review by LynnerKevin — APPROVED** [review]
Matching the real world distribution of languages is probably a bit more than we need for mockup data but otherwise looks right to me

File: notebooks/dan_kositzke_notebook_testing.ipynb [hunk 1]
```diff
@@ -0,0 +1,1651 @@
...
+      ],
+      "metadata": {
+        "id": "v2z0c2MCQo3W"
+      }
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "Assumption: The distribution of the desired programming languages of the mentees and the known programming languages of the mentors will be similar to the distribution of the most used programming languages in 2021."
```
Comment (Cole-Richards) [inline notebooks/dan_kositzke_notebook_testing.ipynb]: This was an excellent idea to control the distribution of programming languages.

File: notebooks/dan_kositzke_notebook_testing.ipynb [hunk 2]
```diff
@@ -0,0 +1,1651 @@
...
+      "metadata": {
+        "id": "7kxIZ6wt5FEz"
+      }
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "def get_unique_languages():\n",
+        "  \"\"\"\n",
+        "  Generates a list of 3 languages for each random mentor.\n",
```
Comment (Cole-Richards) [inline notebooks/dan_kositzke_notebook_testing.ipynb]: When you say random here, are you saying that you're not incorporating the percentages from statista_data? If that is the case, I would encourage consistency between the mentee's and mentor's for more accurate results?

File: notebooks/dan_kositzke_notebook_testing.ipynb [hunk 3]
```diff
@@ -0,0 +1,1651 @@
...
+      "source": [
+        "class RandomMentor:\n",
+        "  def __init__(self):\n",
+        "    self.id = str(random.randint(50000000,100000000))\n",
+        "    self.name = names.get_full_name()\n",
+        "    self.languages = get_unique_languages()\n",
+        "  \n",
+        "  def __repr__(self):\n",
+        "    output = (\n",
+        "        f\"ID: {self.id}\",\n",
```
Comment (Cole-Richards) [inline notebooks/dan_kositzke_notebook_testing.ipynb]: The simplicity of f formatting is something I like seeing: well done.

File: notebooks/dan_kositzke_notebook_testing.ipynb [hunk 4]
```diff
@@ -0,0 +1,1651 @@
...
+          },
+          "metadata": {},
+          "execution_count": 464
+        }
+      ]
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "# Loop through mentees/mentors and match them up\n",
```
Comment (Cole-Richards) [inline notebooks/dan_kositzke_notebook_testing.ipynb]: Clear that machine learning is still not quite necessary with the data provided. I like seeing simplicity where its all that is needed. Well done.

**Review by Cole-Richards — APPROVED** [review]
Like you said, ideally all mentees would be matched with mentors. I think this is a great start with research. Looking forward to how it develops.

**Comment by sysgear** [comment]
Great start. like to see how it develops

**Review by sysgear — APPROVED** [review]
Great start. Like to see how it develops

**Review by jinjahninjah — APPROVED** [review]
I really like the idea of incorporating current, dynamic data on what languages are popular. We could definitely give admin a list of mentees that weren't matched, we can ask them when we meet next.

**Review by Katerynapass — APPROVED** [review]
Great job generating mock data and matching mentees with mentors.

**Pull request merged**